packaging/imagemode/Makefile

```diff
@@ -0,0 +1,139 @@
+
+#
+# Define the main targets
+#
+all:
+	@echo "make [rhocp | repourl | <build_ver> | run | clean]"
+	@echo "   rhocp:       build a MicroShift bootc image using 'rhocp' repository packages"
+	@echo "                with versions specified as 'USHIFT_VER=value'"
+	@echo "   repourl:     build a MicroShift bootc image using custom repository URLs"
+	@echo "                specified as 'USHIFT_URL=value' and 'OCPDEP_URL=value'"
+	@echo "   <build_ver>: build a MicroShift bootc image of a specific version from the"
+	@echo "                available predefined configurations listed below"
+	@echo "   run:         run the local 'microshift-\$${IMAGE_VER}' bootc image version"
+	@echo "                specified as 'IMAGE_VER=value'"
+	@echo "   stop:        stop the local running 'microshift-*' containers"
+	@echo "   clean:       delete all local container images"
+	@echo ""
+	@echo "Available build versions:"
+	@$(MAKE) -qp | awk -F':' '/^4.[^\t]*:([^=]|$$)/ {print "   " $$1}' | sort -u
+
+rhocp:
+ifndef USHIFT_VER
+	$(error Run 'make rhocp USHIFT_VER=value')
+endif
+	IMAGE_VER=$$(sudo dnf repoquery microshift \
+		--quiet \
+		--queryformat '%{version}' \
+		--repo "rhocp-${USHIFT_VER}-for-rhel-9-$$(uname -m)-rpms" \
+		--latest-limit 1) ; \
+	IMAGE_NAME="microshift-$${IMAGE_VER}" ; \
+	sudo podman build \
+		--no-cache \
+		--authfile "${PULL_SECRET}" \
+		--build-arg BASE_IMAGE_URL=${BASE_IMAGE_URL} \
+		--build-arg BASE_IMAGE_TAG=${BASE_IMAGE_TAG} \
+		--build-arg USHIFT_VER=$${USHIFT_VER} \
+		-t "$${IMAGE_NAME}" \
+		-f Containerfile.rhocp
+
+repourl:
+ifndef USHIFT_URL
+	$(error Run 'make repourl USHIFT_URL=value OCPDEP_URL=value')
+endif
+ifndef OCPDEP_URL
+	$(error Run 'make repourl USHIFT_URL=value OCPDEP_URL=value')
+endif
+	IMAGE_VER=$$(sudo dnf repoquery microshift \
+		--quiet \
+		--queryformat '%{version}' \
+		--disablerepo '*' \
+		--repofrompath "this,${USHIFT_URL}" \
+		--latest-limit 1 | sed 's/~/-/g') ; \
+	IMAGE_NAME="microshift-$${IMAGE_VER}" ; \
+	sudo podman build \
+		--no-cache \
+		--authfile "${PULL_SECRET}" \
+		--build-arg BASE_IMAGE_URL=${BASE_IMAGE_URL} \
+		--build-arg BASE_IMAGE_TAG=${BASE_IMAGE_TAG} \
+		--build-arg USHIFT_VER=$${IMAGE_VER} \
+		--build-arg USHIFT_URL=$${USHIFT_URL} \
+		--build-arg OCPDEP_URL=$${OCPDEP_URL} \
+		-t "$${IMAGE_NAME}" \
+		-f Containerfile.repourl
+
+run:
+ifndef IMAGE_VER
+	$(error Run 'make run IMAGE_VER=value')
+endif
+	IMAGE_NAME="microshift-${IMAGE_VER}" ; \
+	sudo modprobe openvswitch ; \
+	CONT_ID=$$(sudo podman run --rm -d --privileged \
+		-v "${PULL_SECRET}":/etc/crio/openshift-pull-secret:ro \
+		-v /var/lib/containers/storage:/var/lib/containers/storage \
+		--replace \
+		--name "$${IMAGE_NAME}" \
+	    "$${IMAGE_NAME}") ; \
+	echo "sudo podman exec -it $${CONT_ID} bash"
+
+stop:
+	CONT_IDS=$$(sudo podman ps --format "{{.Names}}" | grep '^microshift-' || true) ; \
+	if [ -n "$${CONT_IDS}" ] ; then sudo podman stop $${CONT_IDS} ; fi
+
+clean:
+	sudo podman rmi -af
```

Comment:
Can we do something more sophisticated? We cannot assume that this only runs on a microshift dev vm where someone can not care about the images and we can wipe everything

Reply:
Do you have a suggestion what to do?
Probably, we should just remove the `clean` target, which is not very helpful in its current form.

Reply:
We're producing images with names starting with `microshift-` so maybe we could clean only those?